src/core/dispatcher.ts

```diff
@@ -51,6 +52,19 @@ export class Dispatcher implements BindingObserverDelegate {
     this.application.handleError(error, `Error ${message}`, detail)
   }
 
+  private clearEventListenersForBinding(binding: Binding) {
+    const eventListener = this.fetchEventListenerForBinding(binding)
+    if (eventListener.bindings.length == 0) {
+      const { eventTarget, eventName, eventOptions } = binding
+      const eventListenerMap = this.fetchEventListenerMapForEventTarget(eventTarget)
+      const cacheKey = this.cacheKey(eventName, eventOptions)
+      eventListener.disconnect()
+      eventListenerMap.delete(cacheKey)
+
+      if (eventListenerMap.size == 0) this.eventListenerMaps.delete(eventTarget)
+    }
+  }
```

Comment:
Maybe could extract a couple of methods to clarify logic. I also added a method `EventListener.hasBindings`:

```ts
private clearEventListenersForBinding(binding: Binding) {
  const eventListener = this.fetchEventListenerForBinding(binding)
  if (!eventListener.hasBindings()) {
    this.removeMappedEventListenerFor(binding)
    eventListener.disconnect()
  }
}

private removeMappedEventListenerFor(binding: Binding) {
  const { eventTarget, eventName, eventOptions } = binding
  const eventListenerMap = this.fetchEventListenerMapForEventTarget(eventTarget)
  const cacheKey = this.cacheKey(eventName, eventOptions)

  eventListenerMap.delete(cacheKey)
  if (eventListenerMap.size == 0) this.eventListenerMaps.delete(eventTarget)
}
```